pkg/apis/core/helper/helpers.go

```
@@ -167,3 +169,61 @@ func FindWorkerByName(workers []core.Worker, name string) *core.Worker {
+	return getVersionDiff(old, new)
+}
+
+// GetAddedVersions finds versions that have been added in the new compared to the new version slice.
+// returns a map associating the index in the new version slice with the version and a string set containing the versions.
+func GetAddedVersions(old []core.ExpirableVersion, new []core.ExpirableVersion) (map[int]core.ExpirableVersion, sets.String) {
+	return getVersionDiff(new, old)
+}
+
+// getVersionDiff gets versions that are in v1 but not in v2.
+// Returns versions mapped to its index in v1 and also a string set containing the versions
+func getVersionDiff(v1, v2 []core.ExpirableVersion) (map[int]core.ExpirableVersion, sets.String) {
+	v2Versions := make(map[string]struct{}, len(v2))
+	for _, x := range v2 {
+		v2Versions[x.Version] = struct{}{}
+	}
+	diff := make(map[int]core.ExpirableVersion, len(v1))
+	set := sets.String{}
+	for index, x := range v1 {
+		if _, found := v2Versions[x.Version]; !found {
+			diff[index] = x
+			set.Insert(x.Version)
+		}
+	}
+	return diff, set
+}
+
+// FilterVersionsWithClassification filters versions for a classification
+func FilterVersionsWithClassification(versions []core.ExpirableVersion, classification core.VersionClassification) []core.ExpirableVersion {
+	var result []core.ExpirableVersion
+	for _, version := range versions {
+		if version.Classification == nil || *version.Classification != classification {
+			continue
+		}
+		result = append(result, version)
+	}
+	return result
+}
+
+// FindVersionsWithSameMajorMinor filters the given versions slice for versions having the same major and minor version as the given version
+func FindVersionsWithSameMajorMinor(versions []core.ExpirableVersion, version semver.Version) ([]core.ExpirableVersion, error) {
+	var result []core.ExpirableVersion
+	for _, v := range versions {
+		semVer, err := semver.NewVersion(v.Version)
```

I wonder if we can ignore this error since the semver check is done by the validator.

---

Yes makes sense, thanks